tests/integration/devfile/cmd_devfile_test_test.go

```diff
@@ -0,0 +1,114 @@
+		helper.Chdir(currentWorkingDirectory)
+		err := os.Setenv("KUBECONFIG", originalKubeconfig)
+		Expect(err).NotTo(HaveOccurred())
+		helper.DeleteDir(context)
+		os.Unsetenv("GLOBALODOCONFIG")
+	})
+
+	Context("Should show proper errors", func() {
+
+		It("should show error if component is not pushed", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", namespace, cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-testgroup.yaml"), filepath.Join(context, "devfile.yaml"))
+
+			output := helper.CmdShouldFail("odo", "test", "--context", context)
+
+			Expect(output).To(ContainSubstring("component does not exist, a valid component is required to run 'odo test'"))
+		})
+
+		It("should show error for devfile v1", func() {
+			helper.CmdShouldPass("odo", "create", "springBoot", "--context", context, cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "springboot", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfilesV1", "springboot", "devfile-init.yaml"), filepath.Join(context, "devfile.yaml"))
+
+			output := helper.CmdShouldFail("odo", "test", "--context", context)
+
+			Expect(output).To(ContainSubstring("'odo test' is not supported in devfile 1.0.0"))
+		})
+
+		It("should show error if no test group is defined", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--context", context, cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile.yaml"), filepath.Join(context, "devfile.yaml"))
+			helper.CmdShouldPass("odo", "push", "--context", context)
+			output := helper.CmdShouldFail("odo", "test", "--context", context)
+
+			Expect(output).To(ContainSubstring("the command group of kind \"test\" is not found in the devfile"))
+		})
+	})
+
+	Context("Should run test command successfully", func() {
+
+		It("Should run test command successfully with only one default specified", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--context", context, cmpName)
+
+			helper.CopyExample(filepath.Join("source", "devfiles", "nodejs", "project"), context)
+			helper.CopyExampleDevFile(filepath.Join("source", "devfiles", "nodejs", "devfile-with-testgroup.yaml"), filepath.Join(context, "devfile.yaml"))
+			helper.CmdShouldPass("odo", "push", "--context", context)
+
+			output := helper.CmdShouldPass("odo", "test", "--context", context)
+			helper.MatchAllInOutput(output, []string{"Executing test1 command", "mkdir test1"})
+
+			podName := cliRunner.GetRunningPodNameByComponent(cmpName, namespace)
+			output = cliRunner.ExecListDir(podName, namespace, sourcePath)
+			Expect(output).To(ContainSubstring("test1"))
+		})
+
+		It("Should run test command successfully with test-command specified", func() {
```

Comment:
Can you also add a case where you're specifying test command with an invalid command like:
1. wrong name(ie the command is not present)
2. a build/run command(--test-command="some build command")

```
Maysuns-MacBook-Pro:springboot-ex maysun$ odo test --test-command="devrun"

Executing devfile test command for component testm
 ✗  command group mismatched, command devrun is of group run in devfile.yaml
Maysuns-MacBook-Pro:springboot-ex maysun$ 
Maysuns-MacBook-Pro:springboot-ex maysun$ 
Maysuns-MacBook-Pro:springboot-ex maysun$ 
Maysuns-MacBook-Pro:springboot-ex maysun$ odo test --test-command="some wrong command"

Executing devfile test command for component testm
 ✗  the command "some wrong command" is not found in the devfile
```

Probably add the two cases for docker as well

Reply:
added integration tests